iam/api-client/src/main/java/com/google/iam/snippets/ServiceAccountDelete.java

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.iam.snippets;
+
+// [START iam_delete_service_account]
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.iam.v1.Iam;
+import com.google.api.services.iam.v1.IamScopes;
+import java.util.Collections;
+
+
+public class ServiceAccountDelete {
+
+    public static void deleteServiceAccount(String email) throws Exception {
```

Comment:
Avoid throwing Exceptions at the top level, and instead show the user how to properly handle them (using try/catch). See [Exception Handling](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#exception-handling).

Reply:
Is there style guidance for how to deal with exceptions that might be thrown when initializing a variable? For example, in most of these snippets, the possible exception is thrown when building the IAM service, which is also when I initialize the service variable. I need to initialize it outside the block somehow or I can't use it.
Do I just initialize it as null?

Reply:
There are usually 2 options which are viewed as acceptable:

1. You can use null or avoid assignment to keep the scope function wide:
```java
Iam service;
try {
   GoogleCredential credential = GoogleCredential.getApplicationDefault()
        .createScoped(Collections.singleton(IamScopes.CLOUD_PLATFORM));

    Iam service = new Iam.Builder(GoogleNetHttpTransport.newTrustedTransport(), JacksonFactory.getDefaultInstance(),
            credential).setApplicationName("service-account-keys").build();
} catch (IOException ex) {
   // Here is where you handle the bad thing
   System.out.println("Unable to initialize the IAM service: " + e);
   return
}
```

2. If it's a somewhat short bit of code (and there aren't any other throws conflicting) sometimes folks prefer putting the entire block around it all:
```java
try{
   GoogleCredential credential = GoogleCredential.getApplicationDefault()
        .createScoped(Collections.singleton(IamScopes.CLOUD_PLATFORM));

    Iam service = new Iam.Builder(GoogleNetHttpTransport.newTrustedTransport(), JacksonFactory.getDefaultInstance(),
            credential).setApplicationName("service-account-keys").build();

    ServiceAccountKey key = service.projects().serviceAccounts().keys()
            .create("projects/-/serviceAccounts/" + serviceAccountEmail, new 
    CreateServiceAccountKeyRequest()).execute();

    System.out.println("Created key: " + key.getName());
} catch (IOException ex) {
    // Here is where you handle the bad thing
   System.out.println("Unable to initialize the IAM service: " + e);
}
```

Reply:
Done